pyfstat/core.py

```diff
@@ -1588,16 +1588,15 @@ def _init_semicoherent_window_range(self):
         self.semicoherentWindowRange.dtau = int(1)  # Irrelevant
 
     def init_semicoherent_parameters(self):
+        self.endTime = self.maxStartTime + self.Tsft
         logging.info(
             (
                 "Initialising semicoherent parameters from {} to {} in" " {} segments"
-            ).format(self.minStartTime, self.maxStartTime, self.nsegs)
+            ).format(self.minStartTime, self.endTime, self.nsegs)
         )
         self.transientWindowType = "rect"
         self.whatToCompute = lalpulsar.FSTATQ_2F + lalpulsar.FSTATQ_ATOMS_PER_DET
-        self.tboundaries = np.linspace(
-            self.minStartTime, self.maxStartTime, self.nsegs + 1
-        )
+        self.tboundaries = np.linspace(self.minStartTime, self.endTime, self.nsegs + 1)
```

Comment:
Doesn't this contradict Line 1609
```
if self.tboundaries[-1] > self.SFT_timestamps[-1]:
```
since `self.endTime = self.maxStartTime + self.Tsft` and `self.SFT_timestamps[-1] === self.maxStartTime`?

Reply:
Yes, that looks inconsistent. I hadn't checked that "SFT_timestamps" case, let's see if we can add that to the test suite... (insanity ensues)

Reply:
No, wait, that's not a separate user option; `self.SFT_timestamps` always gets set  unless `sftfilepattern==None`, so that block should get triggered in almost all test cases. That's confusing then that it never fails so far... Will look into it.

Reply:
...oh, it's just a `logging.debug`, not an error. And yes it does get triggered by default. Not sure why it's there though, I think this is actually what we expect following lal's conventions:

- SFT timestamps are always the start times of the data stretches used for each one
- segment boundaries say "take all data within"

So in combination, we indeed expect in an ideal situation (no-gaps data, PyFstat splits it into segments itself, with neat integer division working out):
- self.tboundaries[0] == self.SFT_timestamps[0]
- self.tboundaries[-1] = self.SFT_timestamps[-1] + 1800

Reply:
`self.tboundaries` are used in  [pyfstat/core.py/SemiCoherentSearch](https://github.com/PyFstat/PyFstat/blob/4af3c01a988f8633e2024d1cc3522d701702be59/pyfstat/core.py#L1579) to define `semicoherentWindowRange`.  `-self.Tcoh` is included because we want the last starting time.


The following situations define their own `tboundaries` (this is just for the record)
1. [pyfstat/optimal_setup_functions.py/get_Nstar_estimate](https://github.com/PyFstat/PyFstat/blob/master/pyfstat/optimal_setup_functions.py#L232) where it's used consistently (i.e. the last `tboundary` gets used only as end-segment time)

2. [pyfstat/mcmc_based_searches.py/get_p_value](https://github.com/PyFstat/PyFstat/blob/master/pyfstat/mcmc_based_searches.py#L2060) and [pyfstat/mcmc_based_searches.py/plot_cumulative_max](https://github.com/PyFstat/PyFstat/blob/master/pyfstat/mcmc_based_searches.py#L2423), both of which are related to

Reply:
--> #126

Reply:
Not resolving this one here for now, will check tomorrow if this part of the change can be dropped from this PR without breaking the tests, as #126 would just revert it anyway.

Reply:
Essentially without this part of the diff, the SemiCoherentSearch test (or any call to that class without manually adjusting maxStartTime) will fail (or use too little data) because for maxStartTime to be the right choice here, when it was determined from SFTs the manual shift upwards as [here](https://github.com/PyFstat/PyFstat/pull/126/files#diff-ed8f52637a7d535159fd46ef4d8a91a2R715) is needed.

I'm replacing 2a33e7fdcb4463528e1da438f01d417fabd4d81a with a minimal one-line fix to just do just that, with locally defined Tsft, to be then further cleaned up by #125.

Reply:
Simplified commit: 1147c101098a37ade8fa18cf826d466f65ba8dfc